### ISSUE TITLE
Is my solution to key error during training correct?!

### ISSUE THREAD

**Issue by mrkind** [xingyizhou/CenterNet#357]
My own data contains 20 categories excluding "__background__",21 categories including "__background__".

According to DEVELOP.md and this blog https://blog.csdn.net/weixin_41765699/article/details/100118353 .I copied src/lib/datasets/dataset/coco.py and made some modification and form my own file bu.py which is shown below.
![66](https://user-images.githubusercontent.com/14086114/64907943-ffff5580-d72b-11e9-946e-daee19298dcf.png)

At first,according to the blog,line 14 in bu.py was and should be
  num_classes = 20
and line 46 in bu.py was 
  self._valid_ids = list(range(20))
Under these settings I got a key error during training show below.

![55](https://user-images.githubusercontent.com/14086114/64907987-861b9c00-d72c-11e9-962c-5ea1c0a4eb6e.png)

Basically it said the key '20' is out of range so I changed line 14 into 

 num_classes = 21

and changed line 46 into 

self._valid_ids = list(range(21))

Then everything seems ok and training can be performed.However isn't it that the variable num_classes should be the number of classes excluding "__background__"?

Is the modification made by me correct?Will the artificial increase of the number of classes by 1 significantly impact the training result?

**Comment by David-19940718** [comment]
@mrkind Hi, you can just set self._valid_ids = list(range(1, 21))  to fix this problem, it may work.
I follow by [this](https://blog.csdn.net/weixin_43509263/article/details/100799415), you can refer to this blog.

**Comment by mrkind** [comment]
Thank you for providing this guide! I will read it,and I will try your suggestion.

**Comment by mrkind** [comment]
@David-19940718 ,I have verified your suggestion's effectiveness.
